src/js/components/ToggleButton.d.ts

```diff
@@ -0,0 +1,11 @@
+import { Component } from "react";
+
+interface IToggleButtonProps {
+  checked: boolean;
+  children: any;
+  onChange: () => void;
+  checkboxClassName: string[] | object | string;
+  className: string[] | object | string;
+}
+
+export default class ToggleButton extends Component<IToggleButtonProps, {}> {}
```

Comment:
AFAIK you can say `Component<PropsInterface>` and dont need to pass a state type if there is no state - and if there is state, I think we should add it 🤷‍♂️